sql/core/src/main/scala/org/apache/spark/sql/execution/command/functions.scala

```diff
@@ -46,26 +46,47 @@ case class CreateFunctionCommand(
     functionName: String,
     className: String,
     resources: Seq[FunctionResource],
-    isTemp: Boolean)
+    isTemp: Boolean,
+    ifNotExists: Boolean,
+    replace: Boolean)
   extends RunnableCommand {
 
+  if (ifNotExists && replace) {
+    throw new AnalysisException("CREATE FUNCTION with both IF NOT EXISTS and REPLACE" +
+      " is not allowed.")
+  }
+
+  // Disallows 'CREATE TEMPORARY FUNCTION IF NOT EXISTS' to be consistent
+  // with 'CREATE TEMPORARY FUNCTION'
```

Comment:
@cloud-fan 
The logic of the  synax  `CREATE TEMPORAY FUNCTION IF NOT EXISTS` like this
1. If the function is already exists, we do nothing not override.
2. If hte function is not exists , we create a new one.
The logic of the  synax  `CREATE OR REPLACE TEMPORAY FUNCTION` like this
1. If the function is already exists, override it.
2. If hte function is not exists , we create a new one.